examples/response_stream/public/containers/app/pages/page_redux_stream/index.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useRef, FC } from 'react';
+
+import {
+  EuiBadge,
+  EuiButton,
+  EuiCheckbox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+import { cancelStream, startStream } from '@kbn/ml-response-stream/client';
+
+import { RESPONSE_STREAM_API_ENDPOINT } from '../../../../../common/api';
+import {
+  setSimulateErrors,
+  setCompressResponse,
+  setFlushFix,
+} from '../../../../../common/api/redux_stream/options_slice';
+import { reset } from '../../../../../common/api/redux_stream/data_slice';
+
+import { Page } from '../../../../components/page';
+import { useDeps } from '../../../../hooks/use_deps';
+
+import { BarChartRace } from '../../components/bar_chart_race';
+import { getStatusMessage } from '../../components/get_status_message';
+
+import { useAppDispatch, useAppSelector } from './hooks';
+
+export const PageReduxStream: FC = () => {
+  const {
+    core: { http, notifications },
+  } = useDeps();
+
+  const dispatch = useAppDispatch();
+  const { isRunning, isCancelled, errors: streamErrors } = useAppSelector((s) => s.stream);
+  const { progress, entities, errors } = useAppSelector((s) => s.data);
+  const { simulateErrors, compressResponse, flushFix } = useAppSelector((s) => s.options);
+
+  const abortCtrl = useRef(new AbortController());
+
+  const onClickHandler = async () => {
+    if (isRunning) {
+      abortCtrl.current.abort();
+      dispatch(cancelStream());
+    } else {
+      abortCtrl.current = new AbortController();
+      dispatch(reset());
+      dispatch(
+        startStream({
+          http,
+          endpoint: RESPONSE_STREAM_API_ENDPOINT.REDUX_STREAM,
+          apiVersion: '1',
+          abortCtrl,
+          body: { compressResponse, flushFix, simulateErrors },
+        })
+      );
+    }
+  };
+
+  // TODO This approach needs to be adapted as it might miss when error messages arrive bulk.
```

Comment:
Are these TODOs for this PR or just for the readers looking at the example application?

Reply:
Yes, it's been like this before, so the idea is this is good enough for the example, but people picking it up should be aware of the limitation.